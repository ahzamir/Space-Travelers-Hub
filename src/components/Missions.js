import React from 'react';

const Missions = () => (
  <div>
    this is miisions page
  </div>
)

export { Missions };