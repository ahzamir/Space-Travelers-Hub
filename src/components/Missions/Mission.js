import React from 'react';
import { useDispatch } from 'react-redux';
import { joinMissions, leaveMissions } from '../../redux-states/missions/MissionState';

const Mission = ({
  title, description, status, id,
}) => {
  const dispatch = useDispatch();
  return (
    <li className="missions border">
      <h2 className="border missionContents">
        {title}
      </h2>
      <p className="border missionContents missionDescription">
        {description}
      </p>
      <div className="border missionContents missionButtons">
        <button
          type="button"
          className={status ? 'missionStatusActive' : 'missionStatus'}
        >
          {status ? 'Active Member' : 'Not a Member'}
        </button>
      </div>
      <div className="border missionContents missionButtons">
        <button
          type="button"
          className={status ? 'missionLeaveButton' : 'missionButton'}
          onClick={() => { status ? dispatch(leaveMissions(id)) : dispatch(joinMissions(id)); }}
        >
          {status ? 'Leave Mission' : 'Join Mission'}
        </button>
      </div>
    </li>
  );
};

export default Mission;
