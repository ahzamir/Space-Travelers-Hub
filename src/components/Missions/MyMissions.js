import React from 'react';

const MyMissions = ({ title }) => (
  <li className="joinedMission border" data-testid="myMissionsContent">
    {title}
  </li>
);

export default MyMissions;
