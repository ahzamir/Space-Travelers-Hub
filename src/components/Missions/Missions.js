import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getMissions } from '../../redux-states/missions/MissionState';
import Mission from './Mission';

const Missions = () => {
  const dispatch = useDispatch();
  const missions = useSelector((state) => state.missions);
  useEffect(() => {
    dispatch(getMissions());
  }, []);
  return (
    <ul className="missionsLists">
      <li className="missions border">
        <h2 className="border missionContents">Missions</h2>
        <h2 className="border missionContents">Description</h2>
        <h2 className="border missionContents">Status</h2>
        <p className="border missionContents" />
      </li>
      {missions.map((mission) => (
        <Mission
          key={mission.mission_id}
          title={mission.mission_name}
          description={mission.description}
          status={mission.status}
          id={mission.mission_id}
        />
      ))}
    </ul>
  );
};

export default Missions;
