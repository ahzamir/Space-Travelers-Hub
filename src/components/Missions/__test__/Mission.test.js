import React from "react";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import store from "../../../redux-states/configureStore";

import Missions from '../Missions';

describe('Testing Mission component', () => {
  render(<Provider store={store}>
    <Missions />
  </Provider>);
  test('Testing the dom for success rendering elements on Mission component', () => {
    expect(screen.queryByTestId('missionsContent')).toBeInTheDocument();
  });

})