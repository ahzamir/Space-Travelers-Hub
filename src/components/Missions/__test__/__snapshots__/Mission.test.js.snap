// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Mission component renders correctly 1`] = `
<ul
  className="missionsLists"
  data-testid="missionsContent"
>
  <li
    className="missions border"
  >
    <h2
      className="border missionContents"
    >
      Missions
    </h2>
    <h2
      className="border missionContents"
    >
      Description
    </h2>
    <h2
      className="border missionContents"
    >
      Status
    </h2>
    <p
      className="border missionContents"
    />
  </li>
</ul>
`;
